lib/components/component.dart

```diff
@@ -89,4 +98,43 @@ abstract class Component {
 
   /// Get a list of non removed effects
   List<ComponentEffect> get effects => _effectsHandler.effects;
+
+  /// Uses the game passed in to prepare the child component before it is added
+  /// to the list of children
+  void addChild(Game gameRef, Component c) {
+    if (gameRef is BaseGame) {
+      gameRef.prepare(c);
+    }
+    children.add(c);
+  }
+
+  /// This method first calls the passed function on itself and then recursively propagates it to every children
+  /// and grandchildren (and so on) of this component, either until it has propagated through the whole tree or
+  /// if the handler at any point returns false.
+  ///
+  /// This method is important to be used by the engine to propagate actions like rendering, taps, etc,
+  /// but you can call it yourself if you need to apply an action to the whole component chain.
+  /// It will only consider components of type T in the hierarchy, so use T = Component to target everything.
+  bool propagateToChildren<T extends Component>(
+    bool Function(T) handler,
```

Comment:
I love the idea of the handler returning a bool, just not the place where it is.

Reply:
The handler is an argument, where would you put it? Or you mean that you don't agree that `propagateToChildren` should be on `component`?